wrappers/csharp/Managed.cs

```diff
@@ -871,6 +873,44 @@ public static string DecryptWithPasswordAsString(string b64data, string password
             return DecryptWithPasswordAsUtf8String(b64data, password, iterations);
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <param name=""></param>
+        /// <returns></returns>
+        public static long JoinShares(byte[][] shares)
+        {
+            var nbShares = shares.Length;
+            var sharesLength = shares[0].Length;
+            var secretLength = (int)Native.JoinSharesSizeNative((UIntPtr)sharesLength);
+            var secret = new byte[secretLength];
+            var tempShares = shares;
+            var sharesTemp = new byte[nbShares][];
+            Array.Copy(shares, sharesTemp, shares.Length);
+            var pointers = Utils.GetArrayReferences(ref shares, nbShares, secretLength);
+            Array.Copy(sharesTemp, shares, sharesTemp.Length);
+            return Native.JoinSharesNative((UIntPtr)nbShares, (UIntPtr)sharesLength, pointers, secret, (UIntPtr)secretLength);
+        }
```

Comment:
1. You should check that all the shares have the same length before doing anything and throw an exception is this is not the case.
2. You should free all the pointers before returning.
3. You should handle the native exceptions(see other managed methods for examples.
4. The method should return a byte[], not the secret isn't even returned.
5. Why is there still a "sharesTemp"?